### ISSUE TITLE
datacats stop while shell/paster run

### ISSUE THREAD

**Issue by wardi** [datacats/datacats#181]
How should we handle this?

Right now existing shell/paster sessions just lose their db/solr access. Should we detect this case and warn the user? Should we kill running shell/paster containers when stopping postgres/solr? Or should we detect this case and leave solr/postgres running even though the user has asked us to stop them (maybe flagging that they should stop after the last shell/paster exits)

Please discuss :snail: :baby_chick:

**Comment by JackMc** [comment]
Add a -f to ``datacats stop`` to remove any web containers? How do we detect shell containers, though?

**Comment by deniszgonjanin** [comment]
Note: Some paster commands rely on the web container being available - eg. `paster datapusher submit`. Others require the web container to be down - eg. https://github.com/ckan/ckan/issues/2306

**Comment by wardi** [comment]
@deniszgonjanin good point. dang. Can we ever know all the commands that need to have a real site running?

**Comment by deniszgonjanin** [comment]
No, we can't. but what is the problem here? Are multiple shell containers not able to connect to postgres at the same time?

**Comment by deniszgonjanin** [comment]
I mean, for commands that require CKAN to be down - it's up to the user to do a `datacats stop` first. All other commands shouldn't shut down the web container. Especially because some harvesting and other long-running commands are ran using paster

**Comment by wardi** [comment]
what I'm saying is that if I have some shell/paster sessions open, then I run `datacats stop` all those sessions are still open but in a broken state because their db/solr went away

**Comment by deniszgonjanin** [comment]
Oh. Can we just issue a warning if there's a running shell container?
```
You have paster/shell containers running. Stop anyway? y/n
```

**Comment by wardi** [comment]
Yes, Then the next trick is figuring out what containers are paster/shell containers. Should we kill the containers if the user says "y" instead of leaving them there with broken connections?

**Comment by deniszgonjanin** [comment]
Yes. liquidate the containers
